storage/innobase/row/row0merge.cc

```diff
@@ -1646,6 +1646,147 @@ row_geo_field_is_valid(
 	return(true);
 }
 
+
+class charset_converter_t {
+public:
+	charset_converter_t(const dict_table_t* old_table,
+			    const dict_table_t* new_table)
+	{
+		if (dict_table_get_n_cols(old_table)
+		    != dict_table_get_n_cols(new_table)) {
+			return;
+		}
```

Comment:
I think that it would be useful to report errors. Constructors do not allow that.

Why can’t we allow the tables to contain a mismatching number of columns? This could be executed as part of an `ALTER TABLE` that is (instantly) adding or dropping columns.